src/components/v2/appDetails/k8Resource/nodeDetail/NodeDetail.component.tsx

```diff
@@ -1,138 +1,229 @@
-import './nodeDetail.css';
-import SummaryComponent from './NodeDetailTabs/Summary.component';
-import { NavLink, Route, Switch } from 'react-router-dom';
-import { useParams, useRouteMatch, useHistory } from 'react-router';
-import { NodeDetailTab } from './nodeDetail.type';
-import { getNodeDetailTabs } from './nodeDetail.util';
-import { LogSearchTermType, NodeType } from '../../appDetails.type';
-import AppDetailsStore from '../../appDetails.store';
-import { useSharedState } from '../../../utils/useSharedState';
-import IndexStore from '../../index.store';
-
-function NodeDetailComponent({ logSearchTerms, setLogSearchTerms }: LogSearchTermType) {
+import React, { useEffect, useState } from 'react'
+import EventsComponent from './NodeDetailTabs/Events.component'
+import LogsComponent from './NodeDetailTabs/Logs.component'
+import ManifestComponent from './NodeDetailTabs/Manifest.component'
+import TerminalComponent from './NodeDetailTabs/Terminal.component'
+import SummaryComponent from './NodeDetailTabs/Summary.component'
+import { NavLink, Redirect, Route, Switch } from 'react-router-dom'
+import { useParams, useRouteMatch } from 'react-router'
+import { NodeDetailTab } from './nodeDetail.type'
+import { getNodeDetailTabs } from './nodeDetail.util'
+import { NodeDetailPropsType, NodeType } from '../../appDetails.type'
+import AppDetailsStore from '../../appDetails.store'
+import { useSharedState } from '../../../utils/useSharedState'
+import IndexStore from '../../index.store'
+import { getManifestResource } from './nodeDetail.api'
+import { showError } from '../../../../common'
+import MessageUI, { MsgUIType } from '../../../common/message.ui'
+import './nodeDetail.css'
+
+function NodeDetailComponent({
+    loadingResources,
+    isResourceBrowserView,
+    selectedResource,
+    logSearchTerms,
+    setLogSearchTerms,
+}: NodeDetailPropsType) {
     const [applicationObjectTabs] = useSharedState(
         AppDetailsStore.getAppDetailsTabs(),
         AppDetailsStore.getAppDetailsTabsObservable(),
-    );
+    )
     const appDetails = IndexStore.getAppDetails()
-    const params = useParams<{ actionName: string; podName: string; nodeType: string }>();
-    const [tabs, setTabs] = useState([]);
-    const [selectedTabName, setSelectedTabName] = useState('');
-    const { path, url } = useRouteMatch();
-    const history = useHistory();
+    const params = useParams<{ actionName: string; podName: string; nodeType: string; node: string }>()
+    const [tabs, setTabs] = useState([])
+    const [selectedTabName, setSelectedTabName] = useState('')
+    const [resourceContainers, setResourceContainers] = useState([])
+    const [isResourceDeleted, setResourceDeleted] = useState(false)
+    const [fetchingResource, setFetchingResource] = useState(false)
+    const { path, url } = useRouteMatch()
+
+    useEffect(() => {
+        const _tabs = getNodeDetailTabs(params.nodeType as NodeType)
```

Comment:
we should add a check for the dependency

Reply:
done